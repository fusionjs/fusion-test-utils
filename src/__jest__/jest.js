/** Copyright (c) 2018 Uber Technologies, Inc.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @flow
 */

import {mockFunction, test} from '../index';

test('function mocks', assert => {
  const myMock = mockFunction();
  assert.equal(myMock.mock.calls.length, 0);
  myMock();
  assert.equal(myMock.mock.calls.length, 1);
});

test('matchSnapshot', assert => {
  const myObj = {foo: 'bar'};
  assert.matchSnapshot(myObj);
});

test('matchSnapshot with name', assert => {
  const myObj = {foo: 'bar'};
  assert.matchSnapshot(myObj, 'my snapshot name');
});
