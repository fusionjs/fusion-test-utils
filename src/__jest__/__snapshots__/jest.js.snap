// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matchSnapshot 1`] = `
Object {
  "foo": "bar",
}
`;

exports[`matchSnapshot with name: my snapshot name 1`] = `
Object {
  "foo": "bar",
}
`;
